src/taskgraph/run-task/run-task

```diff
@@ -978,6 +985,7 @@ def vcs_checkout_from_args(options):
                 revision,
                 ssh_key_file,
                 ssh_known_hosts_file,
+                options.get("shallow", True),
```

Comment:
Let's default to `False` to start. This could potentially cause a lot of backwards compatibility issues, and I'd rather prove out the feature in a handful of tasks that opt into it first. We can file a follow-up to make shallow clones the default later.